internals/schema_migration.py

```diff
@@ -440,3 +440,40 @@ def get_template_data(self, **kwargs) -> str:
         count += 1
     
     return f'{count} FeatureEntry entities given updated field values.'
+
+class UpdateDeprecatedViews(FlaskHandler):
+
+  def get_template_data(self, **kwargs):
+    """Migrates deprecated feature views fields to their new values"""
+    self.require_cron_header()
+
+    for fe in FeatureEntry.query():
+      if fe.ff_views == MIXED_SIGNALS:
+        fe.ff_views = NO_PUBLIC_SIGNALS
+      if fe.ff_views == PUBLIC_SKEPTICISM:
+        fe.ff_views = OPPOSED
+
+      if fe.safari_views == MIXED_SIGNALS:
+        fe.safari_views = NO_PUBLIC_SIGNALS
+      if fe.safari_views == PUBLIC_SKEPTICISM:
+        fe.safari_views = OPPOSED
+      fe.put()
```

Comment:
It would be better to only call put() on a FeatureEntry that is actually changed by one of the conditions above.   Having that logic will make this go faster, but also greatly reduce the risk of stomping on a change to an entry by some human user at the same time.

Reply:
Agreed. Changed 🙂